### PR TITLE
Fix `conf-ncurses`.

### PR DESCRIPTION
**Pull request by OCamlPro-Henry**
See https://github.com/diml/utop/issues/159

### CODE REVIEW

File: packages/conf-ncurses/conf-ncurses.1/opam [hunk 1]
```diff
@@ -7,6 +7,9 @@ license: "MIT"
 build: [
   ["pkg-config" "ncurses"] { os != "darwin"}
 ]
+depends: [
+  "conf-pkg-config" { os != "darwin"}
```
Comment (samoht) [inline packages/conf-ncurses/conf-ncurses.1/opam]: I'm not sure this kind of filters works for `depends` entries

Comment (dsheets) [inline packages/conf-ncurses/conf-ncurses.1/opam]: Do we need a `conf-osx` package so that you can use something like `("conf-pkg-config" | "conf-osx")`?

Comment (OCamlPro-Henry) [inline packages/conf-ncurses/conf-ncurses.1/opam]: You're right the filter is not allowed. I quickly fixed this by removing the filter, but @dsheets proposition might be a better solution.

Comment (samoht) [inline packages/conf-ncurses/conf-ncurses.1/opam]: I'm fine with both solutions, but just keep in mind that every time you add/remove a dependency to ocamlfind, that triggers a large recompilation for our users, so should pick one solution which works and it well tested :-) following that, just depending on `conf-pkg-config` seems to be the less error-prone solution.

Comment (OCamlPro-Henry) [inline packages/conf-ncurses/conf-ncurses.1/opam]: Sure, probably better to wait for the next update of ocamlfind to fix this. Unless the confusion in https://github.com/diml/utop/issues/159 pops up again.

Comment (OCamlPro-Henry) [inline packages/conf-ncurses/conf-ncurses.1/opam]: I did the audit. The dependency is missing in almost every `conf-*` that would require it. I did a quick patch, but I was afraid of the big change. I was planning doing this slowly while auditing `depext` which are not shared in a `conf-*`.

**Pull request merged**